core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -115,7 +116,11 @@ protected void doRefresh() {
   public void commit(TableMetadata base, TableMetadata metadata) {
     // if the metadata is already out of date, reject it
     if (base != current()) {
-      throw new CommitFailedException("Cannot commit: stale table metadata");
+      if (base != null) {
+        throw new CommitFailedException("Cannot commit: stale table metadata");
+      } else {
+        throw new AlreadyExistsException("Table already exists: %s", tableName());
```

Comment:
I didnt look deeply, but wondering why if caller passes base=null and current is not null , we can tell the table already exists?   It seems it would be a new assumption to all the calling code?

Reply:
When `base` is null, that signals that the commit is trying to create the table and it should not exist. When `current` is not null, that signals that the table already exists. So in order to throw `AlreadyExistsException` from the create transaction, this needs to catch that case.

Reply:
yea makes sense now, thanks.  I guess its same concern as @kbendick :)  Maybe we should add this to @base argument javadoc in the TableOperations could clarify it, but itd be unrelated.

Reply:
I added a comment to make this clear.